docs/USAGE-GUIDE.md

```diff
@@ -791,9 +791,25 @@ Authentication is usually the most expensive part of the handshake. To avoid the
 
 ### Configuring the Trust Store
 
-To validate the peer’s certificate, the local “trust store” must contain a certificate that can authenticate the peer’s certificate.
+To validate the peer’s certificate, a config's local “trust store” must contain a certificate
+that can authenticate the peer’s certificate.
 
-By default, s2n-tls will be initialized with the common trust store locations for the host operating system. To completely override those locations, call `s2n_config_wipe_trust_store()`. To add certificates to the trust store, call `s2n_config_set_verification_ca_location()` or `s2n_config_add_pem_to_trust_store()`.
+Configs created with `s2n_config_new_minimal()` contain empty trust stores, and are not able to
+validate any certificates by default. To add certificates to the trust store, call
+`s2n_config_set_verification_ca_location()` or `s2n_config_add_pem_to_trust_store()`.
+
+Operating systems typically install a set of default system certificates used by TLS clients to
+verify the authenticity of public TLS servers. If s2n-tls is operating as a client connecting to
+such a server, `s2n_config_load_system_certs()` can be used to load the system default certificates
+into the trust store. Alternatively, a config can be created with `s2n_config_new()`, which loads
+the system certificates into the trust store by default.
+
+If s2n-tls is operating as a server, or is a client connecting to a non-public TLS server, the
+certificates issued to public TLS servers should not be trusted. In these scenarios, 
```

Comment:
Isn't there potentially a security concern? Like, if s2n-tls is operating as a private server with a private CA that signed a certificate to a client for mutual auth, but the user accidentally forgot to call `s2n_config_wipe_trust_store` on the server, couldn't a non-trusted client get a certificate signed by a public CA and use this to connect to the private server? 

I was kind of trying to ignore/deprecate `s2n_config_new` because of this risk. It seems safer to always just call `s2n_config_new_minimal` and add system certificates manually if they're needed. But I might be misunderstanding the risk!

Our current documentation does specify that `s2n_config_wipe_trust_store` should be called to override the trust store in both functions that add certs to the trust store, so maybe it's okay to just keep this as the recommendation. I re-wrote the documentation to focus on the performance of `s2n_config_new_minimal`, and kept the existing documentation for the new + wipe + add cert pattern.

Reply:
I added a note about the verify_host callback to the Configuring the Trust Store section!